include/deal.II/matrix_free/matrix_free.templates.h

```diff
@@ -373,6 +373,26 @@ namespace internal
                             (cell->level(), cell->index()));
       }
   }
+
+
+
+  // For serial Triangulations, we simply take all cells.
+  // In this case the subdomain_id does not correspond to the process id
+  // Hence, it is not necessarily constant.
+  template <typename InIterator>
+  void resolve_cell_serial (const InIterator   &cell,
+                            std::vector<std::pair<unsigned int,unsigned int> > &cell_its)
+  {
+    if (cell->has_children())
+      for (unsigned int child=0; child<cell->n_children(); ++child)
+        resolve_cell_serial (cell->child(child), cell_its);
+    else
+      {
+        Assert (cell->active(), ExcInternalError());
+        cell_its.push_back (std::pair<unsigned int,unsigned int>
+                            (cell->level(), cell->index()));
+      }
+  }
```

Comment:
I was thinking about passing `subdomain_id` as `numbers::invalid_subdomain_id` in case we do not want to look into the subdomain id. In other words, you would put the statement along the following lines:
```
      if (cell->has_children())
        for (unsigned int child=0; child<cell->n_children(); ++child)
          resolve_cell (cell->child(child), cell_its,
                        subdomain_id);
      else if (subdomain_id == numbers::invalid_subdomain_id ||
               cell->subdomain_id() == subdomain_id)
        {
          Assert (cell->active(), ExcInternalError());
          cell_its.push_back (std::pair<unsigned int,unsigned int>
                              (cell->level(), cell->index()));
        }
```
Since we use the MPI rank as value for the subdomain id in the parallel case, this would be a safe setting. It would work with an arbitrary `subdomain_id` on the cells because I do not check for the subdomain id in the serial case. Or do you mean another use case?